src/supertux/gameconfig.cpp

```diff
@@ -126,36 +134,37 @@ Config::load()
 #endif
   }
 
-  //menu colors
-  float menubackcolor_r, menubackcolor_g, menubackcolor_b, menubackcolor_a;
-  config_mapping.get("menubackcolor_r", menubackcolor_r, 0.2f);
-  config_mapping.get("menubackcolor_g", menubackcolor_g, 0.3f);
-  config_mapping.get("menubackcolor_b", menubackcolor_b, 0.4f);
-  config_mapping.get("menubackcolor_a", menubackcolor_a, 0.8f);
-  menubackcolor = Color(menubackcolor_r, menubackcolor_g, menubackcolor_b, menubackcolor_a);
-
-  float menufrontcolor_r, menufrontcolor_g, menufrontcolor_b, menufrontcolor_a;
-  config_mapping.get("menufrontcolor_r", menufrontcolor_r, 0.6f);
-  config_mapping.get("menufrontcolor_g", menufrontcolor_g, 0.7f);
-  config_mapping.get("menufrontcolor_b", menufrontcolor_b, 0.8f);
-  config_mapping.get("menufrontcolor_a", menufrontcolor_a, 0.5f);
-  menufrontcolor = Color(menufrontcolor_r, menufrontcolor_g, menufrontcolor_b, menufrontcolor_a);
-
-  float hlcolor_r, hlcolor_g, hlcolor_b, hlcolor_a;
-  config_mapping.get("hlcolor_r", hlcolor_r, 0.6f);
-  config_mapping.get("hlcolor_g", hlcolor_g, 0.7f);
-  config_mapping.get("hlcolor_b", hlcolor_b, 1.f);
-  config_mapping.get("hlcolor_a", hlcolor_a, 1.f);
-  hlcolor = Color(hlcolor_r, hlcolor_g, hlcolor_b, hlcolor_a);
-
-  float editorcolor_r, editorcolor_g, editorcolor_b, editorcolor_a;
-  config_mapping.get("editorcolor_r", editorcolor_r, 0.9f);
-  config_mapping.get("editorcolor_g", editorcolor_g, 0.9f);
-  config_mapping.get("editorcolor_b", editorcolor_b, 1.f);
-  config_mapping.get("editorcolor_a", editorcolor_a, 0.6f);
-  editorcolor = Color(editorcolor_r, editorcolor_g, editorcolor_b, editorcolor_a);
-
-  config_mapping.get("menuroundness", menuroundness, 16.f);
+  // menu colors
+
+  std::vector<float> menubackcolor_, menufrontcolor_, menuhelpbackcolor_, menuhelpfrontcolor_,
+    labeltextcolor_, activetextcolor_, hlcolor_, editorcolor_, editorhovercolor_, editorgrabcolor_;
+
+  boost::optional<ReaderMapping> interface_colors_mapping;
+  if (config_mapping.get("interface_colors", interface_colors_mapping))
+  {
+    menubackcolor = Color(interface_colors_mapping->get("menubackcolor", menubackcolor_) ?
```

Comment:
Doesn't the `mapping->get` method have an overload that takes a default parameter?

Reply:
It did not have that before, but it does now. Please rebase!